src/route/index.ts

```diff
@@ -549,17 +551,48 @@ export async function initApiServer(
             const simulationResponse = (await server.simulateTransaction(
               tx
             )) as SorobanRpc.Api.SimulateTransactionSuccessResponse;
+
             const preparedTransaction = SorobanRpc.assembleTransaction(
               tx,
               simulationResponse
             );
+
+            const built = preparedTransaction.build();
+            const sorobanOp = built
+              .operations[0] as Operation.InvokeHostFunction;
+            const auths = sorobanOp.auth || [];
+
+            for (const auth of auths) {
+              if (
+                auth.credentials().switch() !==
+                xdr.SorobanCredentialsType.sorobanCredentialsSourceAccount()
+              ) {
+                throw new Error(ERROR.ACCOUNT_NOT_SOURCE);
+              }
+              if (
+                auth.credentials().switch() ===
+                  xdr.SorobanCredentialsType.sorobanCredentialsAddress() &&
```

Comment:
Is it possible for this condition to be true? The previous check will throw if the credential is not `sorobanCredentialsSourceAccount` right?

Reply:
ah yeah thanks for pointing that out. This shouldn't be needed at all then, I'll remove it.

Reply:
removed in 2b943fbac28c06eb9add751f3477c109aadcd89f